src/app.rs

```diff
@@ -109,9 +112,53 @@ impl ApplicationConfig {
             enterprise,
         })
     }
+
+    /// Configure the API server, if applicable
+    #[cfg(feature = "api")]
+    pub fn setup_api(&self) -> Option<api::Server> {
+        if self.api.enabled {
+            use std::sync::atomic::AtomicBool;
+
+            let api_server = api::Server::start(
+                self.topology.config(),
+                self.topology.watch(),
+                Arc::<AtomicBool>::clone(&self.topology.running),
+            );
+
+            match api_server {
+                Ok(api_server) => {
+                    emit!(ApiStarted {
+                        addr: self.api.address.unwrap(),
+                        playground: self.api.playground
+                    });
+
+                    Some(api_server)
+                }
+                Err(e) => {
+                    error!("An error occurred that Vector couldn't handle: {}.", e);
+                    let _ = self.graceful_crash_sender.send(());
+                    None
+                }
+            }
+        } else {
+            info!(message="API is disabled, enable by setting `api.enabled` to `true` and use commands like `vector top`.");
```

Comment:
Not introduced by your changes, but I question the need for this log. The `api` is disabled by default, and we don't really log like this for other enable-able features.

It reads to me like advertising, which may not be necessary anymore?